### DISCUSSION TITLE
smart objects with layer effects

### DISCUSSION THREAD

**Discussion by bacmono** [joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script#15] [Q&A] [answered]
Hi,

I've been playing around with the script in the Ps CS6 and stumbled across an issue, if the targeted smart object has a Pattern Overlay on (Layer Style). The issue is, that I end up with that pattern across the whole canvas, rather than it being ‘clipped’ to the input file. Please, see the examples below. 

What I want:
![what I want](https://user-images.githubusercontent.com/101568886/158182148-93dbd1be-5fa2-4c25-954f-6606dfbac469.jpg)

What I get:
![what I get](https://user-images.githubusercontent.com/101568886/158182178-a7426869-f29b-485c-bdb5-0950a39c47ca.jpg)

The mockup works properly when replacing the smart object manually.
Thanks for any help.

Cheers

**Comment by joonaspaakko** [comment, marked as answer]
Can you send me the mockup psd file? My email is in my github profile, if needed. It doesn't have to be that exact file, as long as it can reproduce the issue with it.

**Reply by bacmono** [reply]
Just sent the file to your email. Thank you in advance for looking into it!

**Reply by joonaspaakko** [reply]
I think I might need to take a look at this over the weekend. The first thing I got was the issue talked about in this discussion (https://github.com/joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script/discussions/12) not a big deal... but the issue you're having is going to take some hunting down. 

> About your issue: it's not that it like somehow detaches the texture and just plasters it over the entire document... It's all about what doesn't happen inside the target smartobject. It fails to hide the flattened background layer, making it so that the texture is applied to the entire layer rather than just the silhouette. Though it looks like there's some code before it, right after placing the input file, that kinda causes a jam...

If this is time-sensitive, you could download PS CC trial and run it through that (can't guarantee I'll get it fixed within a week). I have no doubt it would work in CC, since it's the version it's been tested with. And I also just tested it myself in PS CC 2019.

**Reply by bacmono** [reply]
Really like your explanation of the issue, thanks! I also noticed that small delay, just before the texture is applied to the entire layer. 

Now, I tried it in Ps CC 2020 and it worked nicely as expected. Furthermore, I eventually updated to CC (since this issue was kind of the final nail in the CS6's coffin), hence, the scipt's fix for CS6 is no longer necessary for me. If done regardless, I'll gladly try it out! 

Wish you all the best!
Thank you so much for your helpfulness & quick replies!
